plutus-core/plutus-ir/src/PlutusIR/Core/Instance/Pretty/Readable.hs

```diff
@@ -0,0 +1,196 @@
+            sequenceDocM ToTheRight juxtFixity $ \prettyEl ->
+                "unwrap" <+> prettyEl term
+        IWrap _ pat arg term   ->
+            sequenceDocM ToTheRight juxtFixity $ \prettyEl ->
+                "iwrap" <+> prettyEl pat <+> prettyEl arg <+> prettyEl term
+        Error _ ty             ->
+            compoundDocM juxtFixity $ \prettyIn ->
+                "error" <+> braces (prettyIn ToTheRight botFixity ty)
+        (viewLet -> Just (lets, body)) ->
+            compoundDocM binderFixity $ \prettyIn ->
+                let prettyBot x = prettyIn ToTheRight botFixity x
+                    prec NonRec = ""
+                    prec _      = "rec"
+                -- Lay out let-bindings in a layout-sensitive way
+                --
+                -- let !x : t = a
+                -- in
+                -- let !y : t = b
+                -- in foo x y
+                in align (sep [ sep ["let" <> prec r <+> align (vcatHard (prettyBot <$> binds)), "in"]
+                              | (r, binds) <- lets ]) <+> prettyBot body
+        Let{} -> error "The impossible happened. This should be covered by the `viewLet` case above."
+
+instance PrettyConstraints configName tyname name uni fun
+          => PrettyBy (PrettyConfigReadable configName) (Binding tyname name uni fun ann) where
+  prettyBy = inContextM $ \case
+    TermBind _ s vdec t ->
+      -- Layout term bindings in lets like
+      --
+      --  let !a : t = body
+      --
+      -- or
+      --
+      --  let !a : t
+      --       = biggerBody
+      withPrettyAt ToTheRight botFixity $ \prettyBot -> do
+        return $ (bt <> prettyBot vdec) <?> "=" <+> prettyBot t
+      where
+        bt | Strict <- s = "!"
+           | otherwise   = "~"
+    TypeBind _ tydec a ->
+      -- Basically the same as above
+      withPrettyAt ToTheRight botFixity $ \prettyBot -> do
+        return $ prettyBot tydec <?> "=" <+> prettyBot a
+    DatatypeBind _ dt -> prettyM dt
+
+instance PrettyConstraints configName tyname name uni fun
+          => PrettyBy (PrettyConfigReadable configName) (Datatype tyname name uni fun ann) where
+  prettyBy = inContextM $ \case
+    Datatype _ tydec pars name cs -> do
+      -- Layout datatypes as
+      --  data (Maybe :: * -> *) a | match_Maybe where
+      --    Nothing : D a
+      --    Just : a -> D a
+      header <- sequenceDocM ToTheRight juxtFixity $ \prettyEl ->
+                  "data" <+> fillSep (prettyEl tydec : map prettyEl pars) <+> "|" <+> prettyEl name <+> "where"
+      withPrettyAt ToTheRight botFixity $ \prettyBot -> do
+        return $ vcatHard [header, indent 2 (align . vcatHard . map prettyBot $ cs)]
+
+instance PrettyReadableBy configName tyname
+          => PrettyBy (PrettyConfigReadable configName) (TyVarDecl tyname ann) where
+  prettyBy = inContextM $ \case
+    TyVarDecl _ x k -> do
+      showKinds <- view $ prettyConfig . pcrShowKinds
+      withPrettyAt ToTheRight botFixity $ \prettyBot -> do
```

Comment:
why isn't this using `typeBinderDocM` any more?

Reply:
Because `typeBinderDocM` gives you too many brackets as it has to deal with not only the declaration but e.g. the surrounding context (like in a type abstraction).

Reply:
Also, that function allows for more arguments than we need here.